mempool/clist_mempool.go

```diff
@@ -391,6 +401,15 @@ func (mem *CListMempool) resCbFirstTime(
 			postCheckErr = mem.postCheck(tx, r.CheckTx)
 		}
 		if (r.CheckTx.Code == abci.CodeTypeOK) && postCheckErr == nil {
+			// Check mempool isn't full again to reduce the chance of exceeding the
+			// limits.
+			if err := mem.isFull(len(tx)); err != nil {
```

Comment:
Maybe we should wrap this in a mutex to serialize the checks? It will be serialized by the mutex in `addTx()` anyway, so serializing here as well shouldn't incur any additional cost.

Reply:
> serializing here as well shouldn't incur any additional cost.

not sure we will get much by doing that

Reply:
It could cause `missing transaction` that `BroadcastTxSync()` returns a `success` (code = 0) but the `tx` is not added to `mempool`.

Reply:
It could cause missing transaction that BroadcastTxSync() returns a success (code = 0) but the tx is not added to mempool.